### PR TITLE
Add Graph Settings

### PR DESCRIPTION
**Pull request by rudyhuyn**
Taking over @sanderl's PR.

### Description of the changes:
- Add Graph Settings
- Add GraphSettingsViewModel to simplify the UI part of the control and manage all the logic
- Support error states (when minimum is higher than maximum)
- Disable keyboard when the flyout is opened
- Only load GraphSettings after the user clicks on the button to improve launch performance

### How changes were validated:
- Manually

### CODE REVIEW

File: src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+                    <RowDefinition Height="10"/>
+                    <RowDefinition Height="auto"/>
+                </Grid.RowDefinitions>
+                <Grid.ColumnDefinitions>
+                    <ColumnDefinition Width="2*"/>
+                    <ColumnDefinition Width="10"/>
+                    <ColumnDefinition Width="2*"/>
+                </Grid.ColumnDefinitions>
+
+                <TextBox x:Name="SettingsXMin"
```
Comment (joseartrivera) [inline src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml]: Really long input causes the UI to stretch and a scrollbar to appear, I think we need to give these TextBoxes a reasonable max width.

File: src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+                         x:Uid="GraphSettingsXMin"
+                         Style="{x:Bind SelectTextBoxStyle(ViewModel.XError), Mode=OneWay}"
+                         PreviewKeyDown="GridSettingsTextBox_PreviewKeyDown"
+                         Text="{x:Bind ViewModel.XMin, Mode=TwoWay}"/>
+                <TextBox x:Name="SettingsXMax"
```
Comment (joseartrivera) [inline src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml]: Pressing ctrl+c to try and copy a value within any of the text boxes causes a crash

File: src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+            </Grid>
+
+            <TextBlock x:Name="UnitsHeading"
+                       x:Uid="UnitsHeading"
+                       Margin="0,16,0,6"
+                       Style="{StaticResource SubTitleTextBoxStyle}"
+                       AutomationProperties.HeadingLevel="Level2"/>
+
+            <StackPanel Orientation="Horizontal">
+                <RadioButton x:Name="Radians"
```
Comment (joseartrivera) [inline src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml]: Changing from Radians/Degrees/Gradians while the graph is empty appears to have no effect and resets the graph to default settings. This causes the values in the y and x min/max to become out of sync with the actual graph values as they also get reset but the textbox has the old values.

File: src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp [hunk 4]
```diff
@@ -493,3 +496,20 @@ void CalculatorApp::GraphingCalculator::ActiveTracing_KeyUp(Windows::UI::Core::C
         args->Handled = true;
     }
 }
+
+void GraphingCalculator::GraphSettingsButton_Click(Object ^ sender, RoutedEventArgs ^ e)
+{
+    if (m_flyoutGraphSettings == nullptr)
```
Comment (joseartrivera) [inline src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp]: Clicking away from the flyout settings to submit your values never shows possible errors to the user, so it may look like nothing occurred. Would it make sense to have some visual indicator on the icon itself? Like an error badge or something? Maybe not?

File: src/GraphControl/Control/Grapher.h [hunk 5]
```diff
@@ -155,6 +156,105 @@ public
...
+        property double YAxisMax
+        {
+            double get()
+            {
+                return m_graph->GetOptions().GetDefaultXRange().second;
+            }
+            void set(double value)
+            {
+                std::pair<double, double> newValue(YAxisMin, value);
+                m_graph->GetOptions().SetDefaultYRange(newValue);
```
Comment (joseartrivera) [inline src/GraphControl/Control/Grapher.h]: Graph settings all get reset when a new equation is inputted, we need to add a call for all of these properties within the UpdateGraphOptions method to fix this.

File: src/GraphControl/Control/Grapher.h [hunk 6]
```diff
@@ -155,6 +156,105 @@ public
...
+            }
+        }
+
+        property double YAxisMin
+        {
+            double get()
+            {
+                return m_graph->GetOptions().GetDefaultXRange().first;
+            }
+            void set(double value)
```
Comment (joseartrivera) [inline src/GraphControl/Control/Grapher.h]: Is there a reason all of these properties do the business logic in the setter rather than in a callback method like the other properties?

File: src/GraphControl/Control/Grapher.h [hunk 7]
```diff
@@ -155,6 +156,105 @@ public
...
+            {
+                std::pair<double, double> newValue(value, YAxisMax);
+                m_graph->GetOptions().SetDefaultYRange(newValue);
```
Comment (joseartrivera) [inline src/GraphControl/Control/Grapher.h]: We should check if m_graph or m_renderMain is null before doing these operations so that the rest of the community can manipulate these without crashing the app. Here and in the rest of the properties.

File: src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml (hunk 1 above)
Comment (joseartrivera) [inline src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml]: On hover in High Contrast White the foreground of the text is black and background is also black.

File: src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+                <Setter Property="Template">
+                    <Setter.Value>
+                        <ControlTemplate TargetType="RadioButton">
+                            <Grid Name="LayoutRoot" Background="{ThemeResource ToggleButtonBackground}">
+                                <VisualStateManager.VisualStateGroups>
+                                    <VisualStateGroup x:Name="CommonStates">
+                                        <VisualState x:Name="Normal"/>
+                                        <VisualState x:Name="PointerOver">
+                                            <VisualState.Setters>
+                                                <Setter Target="LayoutRoot.Background" Value="{ThemeResource ToggleButtonBackgroundPointerOver}"/>
```
Comment (joseartrivera) [inline src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml]: In high contrast ToggleButtonBackgroundPointerOver is the same as the regular background, is that just the default style?

File: src/Calculator/Calculator.vcxproj [hunk 9]
```diff
@@ -16,7 +16,7 @@
     <ApplicationTypeRevision>10.0</ApplicationTypeRevision>
     <AppxDefaultResourceQualifierUAP_Contrast>black</AppxDefaultResourceQualifierUAP_Contrast>
     <AppxBundle>Always</AppxBundle>
-    <PackageCertificateKeyFile>TemporaryKey.pfx</PackageCertificateKeyFile>
+    <!--<PackageCertificateKeyFile>TemporaryKey.pfx</PackageCertificateKeyFile>-->
```
Comment (joseartrivera) [inline src/Calculator/Calculator.vcxproj]: I think we don't want to commit this

File: src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp [hunk 10]
```diff
@@ -493,3 +496,20 @@ void CalculatorApp::GraphingCalculator::ActiveTracing_KeyUp(Windows::UI::Core::C
...
+        auto graphSettings = ref new GraphingSettings();
+        graphSettings->SetGrapher(this->GraphingControl);
+        m_flyoutGraphSettings = ref new Flyout();
+        m_flyoutGraphSettings->Content = graphSettings;
+        auto that(this);
+        m_flyoutGraphSettings->Closed += ref new Windows::Foundation::EventHandler<Platform::Object ^>(
+            [that](Platform::Object ^, Platform::Object ^) { that->GraphingNumPad->IsEnabled = true; });
+    }
+
+    GraphingNumPad->IsEnabled = false;
```
Comment (joseartrivera) [inline src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp]: Why is this needed? Seems weird that this is the only flyout that does this

File: src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp (hunk 4 above)
Comment (joseartrivera) [inline src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp]: Why doesn't this initialization happen somewhere more central? Would protect against other methods of editing the graph settings in the future such as from a hot key, etc.

File: src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml [hunk 11]
```diff
@@ -0,0 +1,168 @@
...
+                    <ColumnDefinition Width="2*"/>
+                    <ColumnDefinition Width="10"/>
+                    <ColumnDefinition Width="2*"/>
+                </Grid.ColumnDefinitions>
+
+                <TextBox x:Name="SettingsXMin"
+                         x:Uid="GraphSettingsXMin"
+                         Style="{x:Bind SelectTextBoxStyle(ViewModel.XError), Mode=OneWay}"
+                         PreviewKeyDown="GridSettingsTextBox_PreviewKeyDown"
+                         Text="{x:Bind ViewModel.XMin, Mode=TwoWay}"/>
```
Comment (joseartrivera) [inline src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml]: I think this needs to be two way, or have some formatting logic since the texbox accepts input like '5-44' but I am not really sure what the double value is and it never becomes updated in the UI

File: src/GraphControl/Control/Grapher.h [hunk 12]
```diff
@@ -155,6 +156,105 @@ public
...
+
+        property double XAxisMin
+        {
+            double get()
+            {
+                return m_graph->GetOptions().GetDefaultXRange().first;
+            }
+            void set(double value)
+            {
+                std::pair<double, double> newValue(value, XAxisMax);
```
Comment (joseartrivera) [inline src/GraphControl/Control/Grapher.h]: Not directly related to this PR but it makes reproing a crash easier when the axis is set to very large/small numbers

Try setting any of the y/x max and mins to very large or very low numbers, then hover your mouse over a graphed line near those large numbers.

in RenderMain::Render() the call to renderer->GetClosePointData crashes with 'An invalid parameter was passed to a function that considers invalid parameters fatal.'

I think it crashes because the range is a double but the API takes a float for the xValueOut and yValueOut params. We may want to file a bug, but also may want to consider handling the ranges as doubles and limiting their size if we can't get a bug fix.

Comment (joseartrivera) [reply src/GraphControl/Control/Grapher.h]: Actually I don't repro this outside of this branch, so maybe something in this branch caused the crash?

File: src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml (hunk 3 above)
Comment (rudyhuyn) [reply src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml]: Workaround added.

File: src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp (hunk 4 above)
Comment (rudyhuyn) [reply src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp]: I've modified the behavior.

If one of the textbox in in an error state, I prevent the popup to be closed, asking the user to fix the values

File: src/Calculator/Calculator.vcxproj (hunk 9 above)
Comment (rudyhuyn) [reply src/Calculator/Calculator.vcxproj]: 👍

File: src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp (hunk 10 above)
Comment (rudyhuyn) [reply src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp]: We can also use `KeyboardShortcutManager::HonorShortcuts` too but I prefer this way:
- We should probably start using KeyboardAccelerator instead of this custom implementation
-  `KeyboardShortcutManager::HonorShortcuts` prevents us to use shortcuts in the popup

File: src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml (hunk 11 above)
Comment (rudyhuyn) [reply src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml]: I don't understand, can you precise?

File: src/GraphControl/Control/Grapher.h (hunk 12 above)
Comment (rudyhuyn) [reply src/GraphControl/Control/Grapher.h]: 👀

Comment (rudyhuyn) [reply src/GraphControl/Control/Grapher.h]: Yes, we should file a bug, 
`        HRESULT GetClosePointData(float inScreenPointX, float inScreenPointY, int& formulaIdOut, float& xScreenPointOut, float& yScreenPointOut, float& xValueOut, float& yValueOut) override`

It should use double, not float (all other fields use double (xmin, xmax, etc...)

Comment (rudyhuyn) [reply src/GraphControl/Control/Grapher.h]: In the meantime, we should ignore this exception

Comment (rudyhuyn) [reply src/GraphControl/Control/Grapher.h]: Unfortunately, we can't try... catch this exception, it happens on another thread. I'm able to repro the issue with the zoom out function, you need to have an equation and zoom out up to 1e17

File: src/GraphControl/Control/Grapher.h (hunk 6 above)
Comment (rudyhuyn) [reply src/GraphControl/Control/Grapher.h]: because these properties are normal, not dependency properties. The main reason is because they don't "host" the data, they are are a bridge to access to the data

File: src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp (hunk 10 above)
Comment (joseartrivera) [reply src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp]: Why do we need to block shorcuts from the flyout menu? Does it cause it to close or something?

File: src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml (hunk 11 above)
Comment (joseartrivera) [reply src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml]: Try typing '5-44' into one of the textboxes. It is accepted and converted to some double behind the scenes, but the resulting double is never displayed in the UI.

File: src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp (hunk 10 above)
Comment (rudyhuyn) [reply src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp]: to prevent the keys to blink when you press numbers

Comment (joseartrivera) [reply src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp]: @EriWong removed the blinking already and that should be in master, so this shouldn't be necessary anymore :)

File: src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml (hunk 11 above)
Comment (rudyhuyn) [reply src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml]: Fixed! std::stod will stop at the first character it doesn't know how to manage. std::stod("5-44") will stop the parsing after the '5' character for example.

I've updated to code to force stod to fully parse the string

**Comment by rudyhuyn** [comment]
Rebased!

File: src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp (hunk 10 above)
Comment (joseartrivera) [reply src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp]: These buttons don't blink in master, no need to add this code.

Comment (rudyhuyn) [reply src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp]: In this case you are right, this code is no more necessary! Removed!

File: src/CalcViewModel/GraphingCalculator/GraphingSettingsViewModel.h [hunk 13]
```diff
@@ -0,0 +1,291 @@
...
+                {
+                    return;
+                }
+                m_XMax = value;
+                if (m_Graph != nullptr)
+                {
+                    try
+                    {
+                        size_t sz;
+                        auto number = std::stod(value->Data(), &sz);
```
Comment (joseartrivera) [inline src/CalcViewModel/GraphingCalculator/GraphingSettingsViewModel.h]: If I zoom out far enough, and click the graph settings, the numbers are red and I can never close the flyout

File: src/CalcViewModel/GraphingCalculator/GraphingSettingsViewModel.h [hunk 14]
```diff
@@ -0,0 +1,291 @@
...
+        {
+            bool get()
+            {
+                return m_Graph != nullptr && m_Graph->TrigUnitMode == (int)Graphing::EvalTrigUnitMode::Radians;
+            }
+            void set(bool value)
+            {
+                if (value && m_Graph != nullptr && m_Graph->TrigUnitMode != (int)Graphing::EvalTrigUnitMode::Radians)
+                {
+                    m_Graph->TrigUnitMode = (int)Graphing::EvalTrigUnitMode::Radians;
```
Comment (joseartrivera) [inline src/CalcViewModel/GraphingCalculator/GraphingSettingsViewModel.h]: Changing the trig mode appears to change the ranges on the Graph but they are not immediately updated in the flyout

File: src/CalcViewModel/GraphingCalculator/GraphingSettingsViewModel.h [hunk 15]
```diff
@@ -0,0 +1,291 @@
...
+                m_XMin = value;
+                if (m_Graph != nullptr)
+                {
+                    try
+                    {
+                        size_t sz;
+                        auto number = std::stod(value->Data(), &sz);
+                        if (value->Length() == sz)
+                        {
+                            m_Graph->XAxisMin = m_XMinValue = number;
```
Comment (joseartrivera) [inline src/CalcViewModel/GraphingCalculator/GraphingSettingsViewModel.h]: I think this is adding trailing zeroes, setting y-max to 75 turns it into 75.000000 in the textbox, maybe consider running through TrimTrailingZeroes?

Comment (rudyhuyn) [reply src/CalcViewModel/GraphingCalculator/GraphingSettingsViewModel.h]: Done

File: src/GraphControl/Control/Grapher.h [hunk 16]
```diff
@@ -100,6 +102,130 @@ public
...
+                    m_renderMain->RunRenderPass();
+                }
+            }
+        }
+
+        void GetDisplayRanges(double* xMin, double* xMax, double* yMin, double* yMax)
+        {
+            try
+            {
+                m_graph->GetRenderer()->GetDisplayRanges(*xMin, *xMax, *yMin, *yMax);
```
[the remark this answers is not in the document]
Comment (rudyhuyn) [reply src/GraphControl/Control/Grapher.h]: the try/catch wasn't for `m_graph` but for GetDisplayRanges crashing when the value returned are out of range

File: src/GraphControl/Control/Grapher.h [hunk 17]
```diff
@@ -100,6 +102,130 @@ public
...
+            }
+            catch (const std::exception&)
+            {
+                OutputDebugString(L"GetDisplayRanges failed\r\n");
+            }
+        }
+
+        void SetDisplayRanges(double xMin, double xMax, double yMin, double yMax)
```
[the remark this answers is not in the document]
Comment (rudyhuyn) [reply src/GraphControl/Control/Grapher.h]: It happens when you zoom out and set a value lower than the current value displayed, the value passed to `m_renderer->SetDisplayRanges` are the correct ones, We should report this bug to team working on the renderer, but it should not block this PR.

File: src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp [hunk 18]
```diff
@@ -505,3 +508,37 @@ void CalculatorApp::GraphingCalculator::ActiveTracing_KeyUp(Windows::UI::Core::C
...
+
+    m_flyoutGraphSettings->ShowAt(GraphSettingsButton);
+}
+
+void GraphingCalculator::OnSettingsFlyout_Closing(
+    FlyoutBase ^ sender,
+    FlyoutBaseClosingEventArgs ^ args)
+{
+    auto graphingSetting = static_cast<GraphingSettings ^>(static_cast<Flyout ^>(sender)->Content);
+    args->Cancel = graphingSetting->CanBeClose();
```
[the remark this answers is not in the document]
Comment (rudyhuyn) [reply src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp]: Good find! But this is not a bug specific to the Graph Calculator:

- Open Date Calculator
- Open the date selector
- Type Alt+1
==> The Date popup will remain

- Open one of the calculator
- Open the Memory/History popup
- Type Alt + 3
==> the memory popup will remain

- Open the Programmer Calculator
- Open the Bit Shift popup
- Type Alt + 1
==> the Bit Shift popup will remain

We need to modify how we handle the navigation between the features. I will cover these issues in another PR.

File: src/CalcViewModel/GraphingCalculator/GraphingSettingsViewModel.h (hunk 14 above)
Comment (rudyhuyn) [reply src/CalcViewModel/GraphingCalculator/GraphingSettingsViewModel.h]: this should not happen, we call `m_renderer->SetDisplayRanges` with the same value after the user selects a different unit. Fixing SetDisplayRanges will also fix this issue.

File: src/GraphControl/Control/Grapher.h (hunk 17 above)
[the remark this answers is not in the document]
Comment (rudyhuyn) [reply src/GraphControl/Control/Grapher.h]: This is due to the `Grapher.ForceProportionalAxes`, we cannot give the user the freedom to set values and enforce proportional axes. 

2 solutions:
- we don't enforce proportional axes
- we modify the value of the 3 other fields when we modify 1.

@grochocki, your opinion?

File: src/CalcViewModel/GraphingCalculator/GraphingSettingsViewModel.h (hunk 14 above)
Comment (rudyhuyn) [reply src/CalcViewModel/GraphingCalculator/GraphingSettingsViewModel.h]: Can you send me screenshots? From what I see, it doesn't. The issue I see isn't the range but how the graph displays them:
Radian:
image

Degree:
image

Both display the same ranges but the way the graph shows them isn't deterministic

File: src/GraphControl/Control/Grapher.h (hunk 16 above)
[the remark this answers is not in the document]
Comment (rudyhuyn) [reply src/GraphControl/Control/Grapher.h]: Oh! I was expecting the mock to mock this part too. The best solution would be to improve the mock and include a fake IGraph and IGraphingOption, so the community will be able to test graph settings, change values, etc..

File: src/GraphControl/Control/Grapher.h (hunk 17 above)
[the remark this answers is not in the document]
Comment (rudyhuyn) [reply src/GraphControl/Control/Grapher.h]: Solution 2 does not seem possible with the current APIs, in some cases, Grapher won't keep the value you have just typed instead of modifying one of the 3 other values.

File: src/GraphControl/Control/Grapher.h [hunk 19]
```diff
@@ -100,6 +102,130 @@ public
...
+        property double YAxisMax
+        {
+            double get()
+            {
+                return m_graph->GetOptions().GetDefaultXRange().second;
+            }
+            void set(double value)
+            {
+                std::pair<double, double> newValue(YAxisMin, value);
+                m_graph->GetOptions().SetDefaultYRange(newValue);
```
[the remark this answers is not in the document]
Comment (rudyhuyn) [reply src/GraphControl/Control/Grapher.h]: conditional added (but no more necessary, the mock implements GetOptions now

File: src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp [hunk 20]
```diff
@@ -69,12 +71,12 @@ GraphingCalculator::GraphingCalculator()
 
     // OemMinus and OemAdd aren't declared in the VirtualKey enum, we can't add this accelerator XAML-side
     auto virtualKey = ref new KeyboardAccelerator();
-    virtualKey->Key = (VirtualKey)187; //OemMinus key
+    virtualKey->Key = (VirtualKey)187; // OemMinus key
```
Comment (joseartrivera) [inline src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp]: Not related to this PR, but ctrl + '-' and ctrl + '+' do the opposite of what clicking the + and - do.

File: src/CalcViewModel/GraphingCalculator/GraphingSettingsViewModel.h [hunk 21]
```diff
@@ -0,0 +1,297 @@
...
+        {
+            bool get()
+            {
+                return m_Graph != nullptr && m_Graph->TrigUnitMode == (int)Graphing::EvalTrigUnitMode::Radians;
+            }
+            void set(bool value)
+            {
+                if (value && m_Graph != nullptr && m_Graph->TrigUnitMode != (int)Graphing::EvalTrigUnitMode::Radians)
+                {
+                    m_Graph->TrigUnitMode = (int)Graphing::EvalTrigUnitMode::Radians;
```
[the remark this answers is not in the document]
Comment (rudyhuyn) [reply src/CalcViewModel/GraphingCalculator/GraphingSettingsViewModel.h]: This is "expected". The Graph control will automatically reset X and Y ranges when you select a different trigo unit, so we call `RefreshPosition()` in the view model after we set a different unit.

Another solution is to not call RefreshPosition and update the UI with the new XMin, Xmax, YMin and YMax values, but it will be very frustrating for users to set these values again.

[the remark this answers is not in the document]
Comment (joseartrivera) [reply src/CalcViewModel/GraphingCalculator/GraphingSettingsViewModel.h]: Could we set the correct x/y min/max values after the graph is initialized but before we render the graph to avoid the flickering?

File: src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp (hunk 20 above)
Comment (rudyhuyn) [reply src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp]: solved
